src/app/legacy/providers/redeem-provider.tsx

```diff
@@ -74,6 +75,45 @@ export function RedeemProvider(props: { children: any }) {
     [inputToken, inputValue],
   )
 
+  const { data: quoteResult, isFetching: isFetchingQuote } = useQuery({
+    initialData,
+    queryKey: [
+      queryKey,
+      {
+        address,
+        inputToken,
+        inputTokenAmount: inputTokenAmount.toString(),
+        nativeTokenPrice,
+        provider,
+        publicClient,
+      },
+    ],
+    queryFn: async () => {
+      if (!address) return null
```

Comment:
if you move these conditions to the 

```typescript
useQuery({
 enabled: Boolean(address && provider && publicClient)  
})
```
it will stop the queryFn call altogether, however you might want to keep the checks, because sadly typescript won't know that they can't not exist thanks to the enabled flag. Of course you can safely remove them too, and just use optional chaining.

Probably checking address is enough too in enabled, since I think if you have an address, you have all the rest.